core/src/main/scripts/importer/validateData.py

```diff
@@ -3100,14 +3110,49 @@ def processCaseListDirectory(caseListDir, cancerStudyId, logger,
         else:
             stableid_files[stable_id] = case
 
-        sampleIds = meta_dictionary['case_list_ids']
-        sampleIds = set([x.strip() for x in sampleIds.split('\t')])
-        for value in sampleIds:
+        # Check if case list category is valid
+        VALID_CATEGORIES = ['all_cases_in_study',
+                            'all_cases_with_mutation_data',
+                            'all_cases_with_cna_data',
+                            'all_cases_with_log2_cna_data',
+                            'all_cases_with_methylation_data',
+                            'all_cases_with_mrna_array_data',
+                            'all_cases_with_mrna_rnaseq_data',
+                            'all_cases_with_rppa_data',
+                            'all_cases_with_microrna_data',
+                            'all_cases_with_mutation_and_cna_data',
+                            'all_cases_with_mutation_and_cna_and_mrna_data',
+                            'all_cases_with_gsva_data',
+                            'other']
+        if 'case_list_category' in meta_dictionary:
+            if meta_dictionary['case_list_category'] not in VALID_CATEGORIES:
+                logger.error('Invalid case list category',
```

Comment:
when giving an error, please specify in the error message, or in the code documentation just above the error, what would otherwise break in cBioPortal.

Reply:
Done